components/mdns/mdns.c

```diff
@@ -6401,6 +6430,16 @@ esp_err_t mdns_service_subtype_add_for_host(const char *instance_name, const cha
     if (!s) {
         return ESP_ERR_NOT_FOUND;
     }
+
+    mdns_subtype_t *srv_subtype = s->service->subtype;
+    while (srv_subtype) {
+        if (strcmp(srv_subtype->subtype, subtype) == 0) {
+            // The same subtype has already been added
+            return ESP_OK;
```

Comment:
nitpick: Could we return something else if the subtype already exits? Maybe `ESP_ERR_INVALID_ARG` as other mdns APIs?

Reply:
done

Reply:
thanks!